storage/statedb/database_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package statedb
+
+import (
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/storage/database"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var childHash = common.HexToHash("1341655")  // 20190805 in hexadecimal
+var parentHash = common.HexToHash("1343A3F") // 20199999 in hexadecimal
+
+func TestDatabase_Reference(t *testing.T) {
+	memDB := database.NewMemoryDBManager()
+	db := NewDatabaseWithCache(memDB, 128)
+
+	assert.Equal(t, memDB, db.DiskDB())
+	assert.Equal(t, 1, len(db.nodes)) // {} : {}
+
+	db.Reference(childHash, parentHash)
+	assert.Equal(t, 1, len(db.nodes)) // {} : {}
```

Comment:
I don't understand what 36~37 lines are doing. Could you explain it, @ehnuje ?

Reply:
To prove that just calling `Reference` does not have effect without manipulation,

Reply:
Okay. Thank you.